### ISSUE TITLE
pip install madmom fails on Windows

### ISSUE THREAD

**Issue by steiml** [CPJKU/madmom#145]
Hi,

I tried to install madmom as described in the user guide but it fails with a compiler error:
C:\Users\michael.stein>pip install madmom
 madmom/features\beats_crf.c(3832) : error C2065: 'INFINITY' : undeclared identifier
madmom/features\beats_crf.c(3967) : error C2065: 'INFINITY' : undeclared identifier

Full output is pasted below

Best,
Michael

Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Users\michael.stein>pip install madmom
Collecting madmom
  Using cached madmom-0.13.2.tar.gz
Building wheels for collected packages: madmom
  Running setup.py bdist_wheel for madmom ... error
  Complete output from command c:\users\michael.stein\anaconda2\python.exe -u -c "import setuptools, tokenize;__file__='
c:\\users\\michae~1.ste\\appdata\\local\\temp\\pip-build-dijxcx\\madmom\\setup.py';exec(compile(getattr(tokenize, 'open'
, open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d c:\users\michae~1.ste\appdata\local\t
emp\tmp8ropqmpip-wheel- --python-tag cp27:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-2.7
  creating build\lib.win-amd64-2.7\madmom
  copying madmom\processors.py -> build\lib.win-amd64-2.7\madmom
  copying madmom\__init__.py -> build\lib.win-amd64-2.7\madmom
  creating build\lib.win-amd64-2.7\madmom\audio
  copying madmom\audio\ffmpeg.py -> build\lib.win-amd64-2.7\madmom\audio
  copying madmom\audio\filters.py -> build\lib.win-amd64-2.7\madmom\audio
  copying madmom\audio\signal.py -> build\lib.win-amd64-2.7\madmom\audio
  copying madmom\audio\spectrogram.py -> build\lib.win-amd64-2.7\madmom\audio
  copying madmom\audio\stft.py -> build\lib.win-amd64-2.7\madmom\audio
  copying madmom\audio\__init__.py -> build\lib.win-amd64-2.7\madmom\audio
  creating build\lib.win-amd64-2.7\madmom\evaluation
  copying madmom\evaluation\alignment.py -> build\lib.win-amd64-2.7\madmom\evaluation
  copying madmom\evaluation\beats.py -> build\lib.win-amd64-2.7\madmom\evaluation
  copying madmom\evaluation\notes.py -> build\lib.win-amd64-2.7\madmom\evaluation
  copying madmom\evaluation\onsets.py -> build\lib.win-amd64-2.7\madmom\evaluation
  copying madmom\evaluation\tempo.py -> build\lib.win-amd64-2.7\madmom\evaluation
  copying madmom\evaluation\__init__.py -> build\lib.win-amd64-2.7\madmom\evaluation
  creating build\lib.win-amd64-2.7\madmom\features
  copying madmom\features\beats.py -> build\lib.win-amd64-2.7\madmom\features
  copying madmom\features\beats_hmm.py -> build\lib.win-amd64-2.7\madmom\features
  copying madmom\features\notes.py -> build\lib.win-amd64-2.7\madmom\features
  copying madmom\features\onsets.py -> build\lib.win-amd64-2.7\madmom\features
  copying madmom\features\tempo.py -> build\lib.win-amd64-2.7\madmom\features
  copying madmom\features\__init__.py -> build\lib.win-amd64-2.7\madmom\features
  creating build\lib.win-amd64-2.7\madmom\ml
  copying madmom\ml\gmm.py -> build\lib.win-amd64-2.7\madmom\ml
  copying madmom\ml\rnn.py -> build\lib.win-amd64-2.7\madmom\ml
  copying madmom\ml\__init__.py -> build\lib.win-amd64-2.7\madmom\ml
  creating build\lib.win-amd64-2.7\madmom\utils
  copying madmom\utils\midi.py -> build\lib.win-amd64-2.7\madmom\utils
  copying madmom\utils\__init__.py -> build\lib.win-amd64-2.7\madmom\utils
  creating build\lib.win-amd64-2.7\madmom\models
  copying madmom\models\LICENSE -> build\lib.win-amd64-2.7\madmom\models
  copying madmom\models\README.rst -> build\lib.win-amd64-2.7\madmom\models
  creating build\lib.win-amd64-2.7\madmom\models\beats
  creating build\lib.win-amd64-2.7\madmom\models\beats\2013
  copying madmom\models\beats\2013\beats_blstm_1.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
  copying madmom\models\beats\2013\beats_blstm_2.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
  copying madmom\models\beats\2013\beats_blstm_3.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
  copying madmom\models\beats\2013\beats_blstm_4.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
  copying madmom\models\beats\2013\beats_blstm_5.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
  copying madmom\models\beats\2013\beats_blstm_6.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
  copying madmom\models\beats\2013\beats_blstm_7.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
  copying madmom\models\beats\2013\beats_blstm_8.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
  creating build\lib.win-amd64-2.7\madmom\models\notes
  creating build\lib.win-amd64-2.7\madmom\models\notes\2013
  copying madmom\models\notes\2013\notes_brnn.npz -> build\lib.win-amd64-2.7\madmom\models\notes\2013
  creating build\lib.win-amd64-2.7\madmom\models\onsets
  creating build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_brnn_1.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_brnn_2.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_brnn_3.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_brnn_4.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_brnn_5.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_brnn_6.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_brnn_7.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_brnn_8.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_rnn_1.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_rnn_2.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_rnn_3.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_rnn_4.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_rnn_5.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_rnn_6.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_rnn_7.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  copying madmom\models\onsets\2013\onsets_rnn_8.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
  creating build\lib.win-amd64-2.7\madmom\models\onsets\2014
  copying madmom\models\onsets\2014\onsets_brnn_pp_1.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
  copying madmom\models\onsets\2014\onsets_brnn_pp_2.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
  copying madmom\models\onsets\2014\onsets_brnn_pp_3.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
  copying madmom\models\onsets\2014\onsets_brnn_pp_4.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
  copying madmom\models\onsets\2014\onsets_brnn_pp_5.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
  copying madmom\models\onsets\2014\onsets_brnn_pp_6.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
  copying madmom\models\onsets\2014\onsets_brnn_pp_7.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
  copying madmom\models\onsets\2014\onsets_brnn_pp_8.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
  creating build\lib.win-amd64-2.7\madmom\models\patterns
  creating build\lib.win-amd64-2.7\madmom\models\patterns\2013
  copying madmom\models\patterns\2013\ballroom_pattern_3_4.pkl -> build\lib.win-amd64-2.7\madmom\models\patterns\2013
  copying madmom\models\patterns\2013\ballroom_pattern_4_4.pkl -> build\lib.win-amd64-2.7\madmom\models\patterns\2013
  running build_ext
  cythoning madmom/ml/rnn.py to madmom/ml\rnn.c
  building 'madmom.ml.rnn' extension
  creating build\temp.win-amd64-2.7
  creating build\temp.win-amd64-2.7\Release
  creating build\temp.win-amd64-2.7\Release\madmom
  creating build\temp.win-amd64-2.7\Release\madmom\ml
  C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\users
\michael.stein\anaconda2\lib\site-packages\numpy\core\include -Ic:\users\michael.stein\anaconda2\include -Ic:\users\mich
ael.stein\anaconda2\PC /Tcmadmom/ml\rnn.c /Fobuild\temp.win-amd64-2.7\Release\madmom/ml\rnn.obj
  rnn.c
  c:\users\michael.stein\anaconda2\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(12) : Warning Msg
: Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
  C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\amd64\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:c:\user
s\michael.stein\anaconda2\libs /LIBPATH:c:\users\michael.stein\anaconda2\PCbuild\amd64 /LIBPATH:c:\users\michael.stein\a
naconda2\PC\VS9.0\amd64 /EXPORT:initrnn build\temp.win-amd64-2.7\Release\madmom/ml\rnn.obj /OUT:build\lib.win-amd64-2.7\
madmom\ml\rnn.pyd /IMPLIB:build\temp.win-amd64-2.7\Release\madmom/ml\rnn.lib /MANIFESTFILE:build\temp.win-amd64-2.7\Rele
ase\madmom/ml\rnn.pyd.manifest
  rnn.obj : warning LNK4197: export 'initrnn' specified multiple times; using first specification
     Creating library build\temp.win-amd64-2.7\Release\madmom/ml\rnn.lib and object build\temp.win-amd64-2.7\Release\mad
mom/ml\rnn.exp
  cythoning madmom/audio/comb_filters.pyx to madmom/audio\comb_filters.c
  building 'madmom.audio.comb_filters' extension
  creating build\temp.win-amd64-2.7\Release\madmom\audio
  C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\users
\michael.stein\anaconda2\lib\site-packages\numpy\core\include -Ic:\users\michael.stein\anaconda2\include -Ic:\users\mich
ael.stein\anaconda2\PC /Tcmadmom/audio\comb_filters.c /Fobuild\temp.win-amd64-2.7\Release\madmom/audio\comb_filters.obj
  comb_filters.c
  c:\users\michael.stein\anaconda2\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(12) : Warning Msg
: Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
  C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\amd64\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:c:\user
s\michael.stein\anaconda2\libs /LIBPATH:c:\users\michael.stein\anaconda2\PCbuild\amd64 /LIBPATH:c:\users\michael.stein\a
naconda2\PC\VS9.0\amd64 /EXPORT:initcomb_filters build\temp.win-amd64-2.7\Release\madmom/audio\comb_filters.obj /OUT:bui
ld\lib.win-amd64-2.7\madmom\audio\comb_filters.pyd /IMPLIB:build\temp.win-amd64-2.7\Release\madmom/audio\comb_filters.li
b /MANIFESTFILE:build\temp.win-amd64-2.7\Release\madmom/audio\comb_filters.pyd.manifest
  comb_filters.obj : warning LNK4197: export 'initcomb_filters' specified multiple times; using first specification
     Creating library build\temp.win-amd64-2.7\Release\madmom/audio\comb_filters.lib and object build\temp.win-amd64-2.7
\Release\madmom/audio\comb_filters.exp
  cythoning madmom/features/beats_crf.pyx to madmom/features\beats_crf.c
  building 'madmom.features.beats_crf' extension
  creating build\temp.win-amd64-2.7\Release\madmom\features
  C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\users
\michael.stein\anaconda2\lib\site-packages\numpy\core\include -Ic:\users\michael.stein\anaconda2\include -Ic:\users\mich
ael.stein\anaconda2\PC /Tcmadmom/features\beats_crf.c /Fobuild\temp.win-amd64-2.7\Release\madmom/features\beats_crf.obj
  beats_crf.c
  c:\users\michael.stein\anaconda2\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(12) : Warning Msg
: Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
  madmom/features\beats_crf.c(3598) : warning C4244: '=' : conversion from 'Py_ssize_t' to 'int', possible loss of data
  madmom/features\beats_crf.c(3607) : warning C4244: '=' : conversion from 'Py_ssize_t' to 'int', possible loss of data
  madmom/features\beats_crf.c(3832) : error C2065: 'INFINITY' : undeclared identifier
  madmom/features\beats_crf.c(3832) : warning C4244: 'initializing' : conversion from 'int' to 'float', possible loss of
 data
  madmom/features\beats_crf.c(3903) : warning C4244: '=' : conversion from 'double' to 'float', possible loss of data
  madmom/features\beats_crf.c(3967) : error C2065: 'INFINITY' : undeclared identifier
  madmom/features\beats_crf.c(19964) : warning C4293: '<<' : shift count negative or too big, undefined behavior
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 9.0\\VC\\BIN\\amd64\\cl.exe' failed with exit status
2

  ----------------------------------------
  Failed building wheel for madmom
  Running setup.py clean for madmom
Failed to build madmom
Installing collected packages: madmom
  Running setup.py install for madmom ... error
    Complete output from command c:\users\michael.stein\anaconda2\python.exe -u -c "import setuptools, tokenize;__file__
='c:\\users\\michae~1.ste\\appdata\\local\\temp\\pip-build-dijxcx\\madmom\\setup.py';exec(compile(getattr(tokenize, 'ope
n', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record c:\users\michae~1.ste\appdata\loc
al\temp\pip-aeif22-record\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-2.7
    creating build\lib.win-amd64-2.7\madmom
    copying madmom\processors.py -> build\lib.win-amd64-2.7\madmom
    copying madmom\__init__.py -> build\lib.win-amd64-2.7\madmom
    creating build\lib.win-amd64-2.7\madmom\audio
    copying madmom\audio\ffmpeg.py -> build\lib.win-amd64-2.7\madmom\audio
    copying madmom\audio\filters.py -> build\lib.win-amd64-2.7\madmom\audio
    copying madmom\audio\signal.py -> build\lib.win-amd64-2.7\madmom\audio
    copying madmom\audio\spectrogram.py -> build\lib.win-amd64-2.7\madmom\audio
    copying madmom\audio\stft.py -> build\lib.win-amd64-2.7\madmom\audio
    copying madmom\audio\__init__.py -> build\lib.win-amd64-2.7\madmom\audio
    creating build\lib.win-amd64-2.7\madmom\evaluation
    copying madmom\evaluation\alignment.py -> build\lib.win-amd64-2.7\madmom\evaluation
    copying madmom\evaluation\beats.py -> build\lib.win-amd64-2.7\madmom\evaluation
    copying madmom\evaluation\notes.py -> build\lib.win-amd64-2.7\madmom\evaluation
    copying madmom\evaluation\onsets.py -> build\lib.win-amd64-2.7\madmom\evaluation
    copying madmom\evaluation\tempo.py -> build\lib.win-amd64-2.7\madmom\evaluation
    copying madmom\evaluation\__init__.py -> build\lib.win-amd64-2.7\madmom\evaluation
    creating build\lib.win-amd64-2.7\madmom\features
    copying madmom\features\beats.py -> build\lib.win-amd64-2.7\madmom\features
    copying madmom\features\beats_hmm.py -> build\lib.win-amd64-2.7\madmom\features
    copying madmom\features\notes.py -> build\lib.win-amd64-2.7\madmom\features
    copying madmom\features\onsets.py -> build\lib.win-amd64-2.7\madmom\features
    copying madmom\features\tempo.py -> build\lib.win-amd64-2.7\madmom\features
    copying madmom\features\__init__.py -> build\lib.win-amd64-2.7\madmom\features
    creating build\lib.win-amd64-2.7\madmom\ml
    copying madmom\ml\gmm.py -> build\lib.win-amd64-2.7\madmom\ml
    copying madmom\ml\rnn.py -> build\lib.win-amd64-2.7\madmom\ml
    copying madmom\ml\__init__.py -> build\lib.win-amd64-2.7\madmom\ml
    creating build\lib.win-amd64-2.7\madmom\utils
    copying madmom\utils\midi.py -> build\lib.win-amd64-2.7\madmom\utils
    copying madmom\utils\__init__.py -> build\lib.win-amd64-2.7\madmom\utils
    creating build\lib.win-amd64-2.7\madmom\models
    copying madmom\models\LICENSE -> build\lib.win-amd64-2.7\madmom\models
    copying madmom\models\README.rst -> build\lib.win-amd64-2.7\madmom\models
    creating build\lib.win-amd64-2.7\madmom\models\beats
    creating build\lib.win-amd64-2.7\madmom\models\beats\2013
    copying madmom\models\beats\2013\beats_blstm_1.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
    copying madmom\models\beats\2013\beats_blstm_2.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
    copying madmom\models\beats\2013\beats_blstm_3.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
    copying madmom\models\beats\2013\beats_blstm_4.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
    copying madmom\models\beats\2013\beats_blstm_5.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
    copying madmom\models\beats\2013\beats_blstm_6.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
    copying madmom\models\beats\2013\beats_blstm_7.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
    copying madmom\models\beats\2013\beats_blstm_8.npz -> build\lib.win-amd64-2.7\madmom\models\beats\2013
    creating build\lib.win-amd64-2.7\madmom\models\notes
    creating build\lib.win-amd64-2.7\madmom\models\notes\2013
    copying madmom\models\notes\2013\notes_brnn.npz -> build\lib.win-amd64-2.7\madmom\models\notes\2013
    creating build\lib.win-amd64-2.7\madmom\models\onsets
    creating build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_brnn_1.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_brnn_2.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_brnn_3.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_brnn_4.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_brnn_5.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_brnn_6.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_brnn_7.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_brnn_8.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_rnn_1.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_rnn_2.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_rnn_3.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_rnn_4.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_rnn_5.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_rnn_6.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_rnn_7.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    copying madmom\models\onsets\2013\onsets_rnn_8.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2013
    creating build\lib.win-amd64-2.7\madmom\models\onsets\2014
    copying madmom\models\onsets\2014\onsets_brnn_pp_1.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
    copying madmom\models\onsets\2014\onsets_brnn_pp_2.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
    copying madmom\models\onsets\2014\onsets_brnn_pp_3.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
    copying madmom\models\onsets\2014\onsets_brnn_pp_4.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
    copying madmom\models\onsets\2014\onsets_brnn_pp_5.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
    copying madmom\models\onsets\2014\onsets_brnn_pp_6.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
    copying madmom\models\onsets\2014\onsets_brnn_pp_7.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
    copying madmom\models\onsets\2014\onsets_brnn_pp_8.npz -> build\lib.win-amd64-2.7\madmom\models\onsets\2014
    creating build\lib.win-amd64-2.7\madmom\models\patterns
    creating build\lib.win-amd64-2.7\madmom\models\patterns\2013
    copying madmom\models\patterns\2013\ballroom_pattern_3_4.pkl -> build\lib.win-amd64-2.7\madmom\models\patterns\2013
    copying madmom\models\patterns\2013\ballroom_pattern_4_4.pkl -> build\lib.win-amd64-2.7\madmom\models\patterns\2013
    running build_ext
    skipping 'madmom/ml\rnn.c' Cython extension (up-to-date)
    building 'madmom.ml.rnn' extension
    creating build\temp.win-amd64-2.7
    creating build\temp.win-amd64-2.7\Release
    creating build\temp.win-amd64-2.7\Release\madmom
    creating build\temp.win-amd64-2.7\Release\madmom\ml
    C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\use
rs\michael.stein\anaconda2\lib\site-packages\numpy\core\include -Ic:\users\michael.stein\anaconda2\include -Ic:\users\mi
chael.stein\anaconda2\PC /Tcmadmom/ml\rnn.c /Fobuild\temp.win-amd64-2.7\Release\madmom/ml\rnn.obj
    rnn.c
    c:\users\michael.stein\anaconda2\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(12) : Warning M
sg: Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
    C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\amd64\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:c:\us
ers\michael.stein\anaconda2\libs /LIBPATH:c:\users\michael.stein\anaconda2\PCbuild\amd64 /LIBPATH:c:\users\michael.stein
\anaconda2\PC\VS9.0\amd64 /EXPORT:initrnn build\temp.win-amd64-2.7\Release\madmom/ml\rnn.obj /OUT:build\lib.win-amd64-2.
7\madmom\ml\rnn.pyd /IMPLIB:build\temp.win-amd64-2.7\Release\madmom/ml\rnn.lib /MANIFESTFILE:build\temp.win-amd64-2.7\Re
lease\madmom/ml\rnn.pyd.manifest
    rnn.obj : warning LNK4197: export 'initrnn' specified multiple times; using first specification
       Creating library build\temp.win-amd64-2.7\Release\madmom/ml\rnn.lib and object build\temp.win-amd64-2.7\Release\m
admom/ml\rnn.exp
    skipping 'madmom/audio\comb_filters.c' Cython extension (up-to-date)
    building 'madmom.audio.comb_filters' extension
    creating build\temp.win-amd64-2.7\Release\madmom\audio
    C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\use
rs\michael.stein\anaconda2\lib\site-packages\numpy\core\include -Ic:\users\michael.stein\anaconda2\include -Ic:\users\mi
chael.stein\anaconda2\PC /Tcmadmom/audio\comb_filters.c /Fobuild\temp.win-amd64-2.7\Release\madmom/audio\comb_filters.ob
j
    comb_filters.c
    c:\users\michael.stein\anaconda2\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(12) : Warning M
sg: Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
    C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\amd64\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:c:\us
ers\michael.stein\anaconda2\libs /LIBPATH:c:\users\michael.stein\anaconda2\PCbuild\amd64 /LIBPATH:c:\users\michael.stein
\anaconda2\PC\VS9.0\amd64 /EXPORT:initcomb_filters build\temp.win-amd64-2.7\Release\madmom/audio\comb_filters.obj /OUT:b
uild\lib.win-amd64-2.7\madmom\audio\comb_filters.pyd /IMPLIB:build\temp.win-amd64-2.7\Release\madmom/audio\comb_filters.
lib /MANIFESTFILE:build\temp.win-amd64-2.7\Release\madmom/audio\comb_filters.pyd.manifest
    comb_filters.obj : warning LNK4197: export 'initcomb_filters' specified multiple times; using first specification
       Creating library build\temp.win-amd64-2.7\Release\madmom/audio\comb_filters.lib and object build\temp.win-amd64-2
.7\Release\madmom/audio\comb_filters.exp
    skipping 'madmom/features\beats_crf.c' Cython extension (up-to-date)
    building 'madmom.features.beats_crf' extension
    creating build\temp.win-amd64-2.7\Release\madmom\features
    C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\use
rs\michael.stein\anaconda2\lib\site-packages\numpy\core\include -Ic:\users\michael.stein\anaconda2\include -Ic:\users\mi
chael.stein\anaconda2\PC /Tcmadmom/features\beats_crf.c /Fobuild\temp.win-amd64-2.7\Release\madmom/features\beats_crf.ob
j
    beats_crf.c
    c:\users\michael.stein\anaconda2\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(12) : Warning M
sg: Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
    madmom/features\beats_crf.c(3598) : warning C4244: '=' : conversion from 'Py_ssize_t' to 'int', possible loss of dat
a
    madmom/features\beats_crf.c(3607) : warning C4244: '=' : conversion from 'Py_ssize_t' to 'int', possible loss of dat
a
    madmom/features\beats_crf.c(3832) : error C2065: 'INFINITY' : undeclared identifier
    madmom/features\beats_crf.c(3832) : warning C4244: 'initializing' : conversion from 'int' to 'float', possible loss
of data
    madmom/features\beats_crf.c(3903) : warning C4244: '=' : conversion from 'double' to 'float', possible loss of data
    madmom/features\beats_crf.c(3967) : error C2065: 'INFINITY' : undeclared identifier
    madmom/features\beats_crf.c(19964) : warning C4293: '<<' : shift count negative or too big, undefined behavior
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 9.0\\VC\\BIN\\amd64\\cl.exe' failed with exit statu
s 2

    ----------------------------------------
Command "c:\users\michael.stein\anaconda2\python.exe -u -c "import setuptools, tokenize;__file__='c:\\users\\michae~1.st
e\\appdata\\local\\temp\\pip-build-dijxcx\\madmom\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read
().replace('\r\n', '\n'), __file__, 'exec'))" install --record c:\users\michae~1.ste\appdata\local\temp\pip-aeif22-recor
d\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in c:\users\michae~1.ste\ap
pdata\local\temp\pip-build-dijxcx\madmom\

C:\Users\michael.stein>

**Comment by superbock** [comment]
Hi,

unfortunately, I can't really help, since I don't have a windows around (and we don't support it officially). However, this seems to be a bug related to cython and/or the compiler version you are using. https://github.com/kivy/pyjnius/issues/214#issuecomment-203702203 seems to have solved a similar (at least to me) issue by upgrading cython via pip. Hope this helps...

**Comment by steiml** [comment]
Thx,

I tried with various older versions of cython (0.22, 0.23) as well as with the most recent one but keep getting the same error. If Windows is not supported than that's the way it is. I'll try on other machines to track down the cause

**Comment by superbock** [comment]
Did you try a more recent compiler version as well? Microsoft Visual Studio 9.0 is from 2008.
It would be nice to have a working windows setup ;)

**Comment by steiml** [comment]
I'm afraid I have bad news. 
I have several versions of Visual Studio / Windows SDK installed but VC9.0 is the one he choses. To force him to use something else, I uninstalled all related products and get the following result when calling "pip install madmom":
error: Microsoft Visual C++ 9.0 is required (Unable to find vcvarsall.bat). Get it from http://aka.ms/vcpython27
Thus, python 2.7 relies / depends on VC9.0.

After installing Microsoft Visual C++ Compiler for Python 2.7 from  http://aka.ms/vcpython27 as suggested in http://stackoverflow.com/questions/2817869/error-unable-to-find-vcvarsall-bat, 
"pip install madmom" returns:
[...]
c:\users\michael.stein\anaconda2\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(12) : Warning Msg: Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
madmom/features\beats_crf.c(3598) : 
	warning C4244: '=' : conversion from 'Py_ssize_t' to 'int', possible loss of data
madmom/features\beats_crf.c(3607) : 
	warning C4244: '=' : conversion from 'Py_ssize_t' to 'int', possible loss of data
madmom/features\beats_crf.c(3832) : 
	error C2065: 'INFINITY' : undeclared identifier
madmom/features\beats_crf.c(3832) : 
	warning C4244: 'initializing' : conversion from 'int' to 'float', possible loss of data
madmom/features\beats_crf.c(3903) : 
	warning C4244: '=' : conversion from 'double' to 'float', possible loss of data
madmom/features\beats_crf.c(3967) : 
	error C2065: 'INFINITY' : undeclared identifier
madmom/features\beats_crf.c(19964) : 
	warning C4293: '<<' : shift count negative or too big, undefined behavior
error: command 'C:\\Users\\michael.stein\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\amd64\\cl.exe' failed with exit status 2

These are the same errors and warnings as before. So my conclusion is that the package is, unfortunately, not windows compliant - at least not with Python 2.7. I don't have a 3.x python to try out whether it would build in this environment. It seems to me that, whatever statement it is that causes the "error C2065: 'INFINITY' : undeclared identifier" stems from generated code that is not compliant with how the compiler used by VC9.0 interprets the standard. It could be a language feature that was only introduced in a later standard or one of the Microsoft quirks where they interpret the standard their way. 

Maybe it's possible (and potentially easiest) to adapt the corresponding python code to make the package usable in windows environments?

**Comment by superbock** [comment]
Ok, didn't know that Cython requires Microsoft Visual C++ 9.0.

Of course it would be best/easiest to fix it in the Python code -- if possible. Maybe replacing our `from libc.math cimport INFINITY` with:
```
cdef extern from "math.h":
    float INFINITY
```
works (source: http://stackoverflow.com/a/16050591). Later in the same thread (http://stackoverflow.com/a/27364128) it is mentioned that  could be also replaced by:
```
from numpy.math cimport INFINITY
```
Maybe one of these solutions work.

**Comment by steiml** [comment]
the second option works for me, the first doesn't. However it is a bit more difficult, because Microsoft is not C-compliant, hence, the stdint.h is missing. I could get it building with the steps below, but that involved downloading the missing header file and putting it in the right place (not very user friendly). From the links provided it seems that a platform switch and typedefs for the data types needed might make it work for everyone. Here's the sequence of steps that made it work for me:

```
git clone --recursive --depth 1 https://github.com/CPJKU/madmom.git
cd madmom
python setup.py develop --user
```

madmom/features\beats_crf.c(3832) : error C2065: 'INFINITY' : undeclared identifier

in madmom/features\beats_crf.pyx
replace [25]
`from libc.math cimport INFINITY`
with 
`from numpy.math cimport INFINITY`
issue solved

`python setup.py develop --user`

madmom/ml\hmm.c(268) : fatal error C1083: Cannot open include file: 'stdint.h': No such file or directory
see:
https://github.com/FSX/misaka/issues/49
https://stackoverflow.com/questions/126279/c99-stdint-h-header-and-ms-visual-studio
https://bugs.launchpad.net/lxml/+bug/1274413

download the header from https://code.google.com/archive/p/msinttypes/ and put it in the right location,i.e. where the compiler searches for it - in my case:
"C:\Users\michael.stein\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\include\stdint.h"

madmom/ml\hmm.c(5339) : error C2065: 'INFINITY' : undeclared identifier
--> known issue, same as in beats_crf.pyx

in madmom/ml\hmm.pyx
replace [24]
`from libc.math cimport INFINITY`
with 
`from numpy.math cimport INFINITY`

Now it builds

**Comment by superbock** [comment]
Ok, so replacing all `from libc.math cimport INFINITY` with `from numpy.math cimport INFINITY` solves the infinity problem.

The `stdint.h` problem stems from defining the back tracking array of the viterbi variables as `uint_32`. This decision was made because of memory reasons.

Using `uint_32` instead of `unsigned int` was needed since `unsigned int` could be translated by the compiler to whatever length an unsigned integer has on that platform. Thus we changed to `uint_32` (https://github.com/CPJKU/madmom/pull/88). However, using
```
cdef np.uint32_t [:, ::1] bt_pointers  = np.empty((num_observations,
                                                          num_states),
                                                         dtype=np.uint32)
```
in line 364 in `hmm.pyx` works on my system. But I fear it will not work for all platforms, since `np.uint32_t` translates to `npy_uint32` (line 734 of https://github.com/cython/cython/blob/master/Cython/Includes/numpy/__init__.pxd) which in turn is `unsigned int` (line 325), which was the root of the problem... @craffel could you maybe check if the above works on your system, although I fear it will not?

So one workaround is manually download the missing header file and put it in the right place. I'm not sure if this is the best, but at least it is working... Let me know if you know of any better one!

**Comment by superbock** [comment]
This works on my machine(s) as well:
```
cdef np.ndarray[np.uint32_t, ndim=2] bt_pointers = \
            np.empty((num_observations, num_states), dtype=np.uint32)
```
Again, I'm not sure if it solves the problem on other platforms. 

@craffel maybe you can try this, too :)

**Comment by superbock** [comment]
`ctypedef np.uint32_t uint32_t` works for building on Windows (see #150). However, the tests fail to run since they can't find the compiled/installed madmom...

**Comment by steiml** [comment]
as pointed out in [https://stackoverflow.com/questions/126279/c99-stdint-h-header-and-ms-visual-studio](https://stackoverflow.com/questions/126279/c99-stdint-h-header-and-ms-visual-studio) something along the lines of this could be a viable workaround:

```
#ifdef _MSC_VER

typedef __int32 int32_t;
typedef unsigned __int32 uint32_t;
typedef __int64 int64_t;
typedef unsigned __int64 uint64_t;

#else
#include <stdint.h>
#endif
```

**Comment by superbock** [comment]
But if I'm not wrong, this has to be included in the `.c` file. So it would involve manual patching/editing. This is not really what we should aim for. The solution mentioned 2 comments above does work, I am however not sure if it breaks #88 again.

**Comment by craffel** [comment]
> @craffel could you maybe check if the above works on your system, although I fear it will not?

Can you put your proposed changes in a branch I can clone to try it out?  Not totally confident I will follow the recipe correctly.

**Comment by superbock** [comment]
Of course, here you go. The branch is named [`test_uint32`](https://github.com/CPJKU/madmom/tree/test_uint32). It contains 3 commits which should work -- or not.

* 0832e34c310adda4aa657628771107490b4e98e0 to confirm the problem still exists
* 7131fc744fa6b092ae107006064952c490b911ff the original fix
* 2eb236722f9f3690b92d7622862073c2a4cb816d to proposed fix

@craffel, thanks for testing!

**Comment by craffel** [comment]
LGTM on the system I had originally been experiencing the problem on.

**Comment by superbock** [comment]
Just to get it correct: with the first commit you still experience out of index errors, while the latter two work?

**Comment by craffel** [comment]
Hmm, yes, I can't reproduce the error at 0832e34.

**Comment by superbock** [comment]
Just to be sure, you recompiled the extensions, right?

**Comment by craffel** [comment]
I removed my old installed madmom, I did a fresh clone and ran `python setup.py install`, which should do the trick, right?  All of the installed binaries were touched, etc.

**Comment by superbock** [comment]
Yes. After checking out the individual commits you have to repeat the install steps. This should result in a Cython run to convert the `.pyx` to `.c` files, followed by compilation.

Seems like the problem disappeared on your machine as it did on mine... Anyways, thanks a lot for testing!

**Comment by superbock** [comment]
I'm closing this issue now, since there is at least some support for Windows now. Please see [this WIKI page](https://github.com/CPJKU/madmom/wiki/Install-on-Windows) for a working setup.

`pip install madmom` will be supported on Windows as soon as [v0.14](https://github.com/CPJKU/madmom/milestone/2) is out.

**Comment by superbock** [comment]
It looks like we have to postpone the ability to`pip install madmom` on Windows to version [v0.15](https://github.com/CPJKU/madmom/milestone/3) or later until #178 is fixed.
